### PR TITLE
Checkstyle

### LINKED ISSUE #317: Inconsistent import order.
The import order in the file `mport Order` is different to the import order checkstyle expects.

E.g. the formatter puts `org` before `java` and checkstyle is not amused about it.

### LINKED ISSUE #318: Checkstyle whitespace
image

Checkstyle is complaining about the whitespace after `)`. I think this whitespace is good here.

### LINKED ISSUE #320: Checkstyle intendation of }
image

Whats the problem here? Checkstyle says: `'}' in Spalte 4 sollte auf der gleichen Linie wie der nächste Teil einer Multi-Block-Anweisung sein.`

### LINKED ISSUE #324: [Checkstyle] Variable Name Minimum Length
Is there a reason for this rule? It's especially confusing to me that Checkstyle seems to be okay with variable `description`.

### LINKED ISSUE #326: [Checkstyle] Allow `protected` Attributes
Checkstyle currently doesn't allow `protected` attributes. I think these should be allowed.

### LINKED ISSUE #327: [Checkstyle] Same String in Different Methods
Is it possible to allow having the same string twice in the same class but in different methods? They can very well have different meanings, so creating a single variable for them both is a bad idea.

### LINKED ISSUE #328: Formatter and Checkstyle Mismatch regarding Indentation

### LINKED ISSUE #332: [Checkstyle] Javadoc summary maximum length
image
Message: Summary too long (max 100 letters).

My JavaDoc is too long. Seriously?

### LINKED ISSUE #334: [Checkstyle] Class Data Abstraction Coupling
So I was just trying to resolve my #325 problem when resolving it gave me this "funny" surprise:

Apparently I'm not allowed to have enough ints in a class to resolve magic number problems of a really basic class which doesn't even span 250 lines and only contains stuff absolutely necessary to produce this single page of a wizard. I have virtually 0 ideas how to resolve this, other than creating an array (the length of which I have to specify with a variable, so that's 1) which I dynamically fill with integers form -10 (another variable) to 10 (a third one) in a loop for which I'll use an inhumanely long counting variable not to violate that rule, so that I have the numbers -10 to 10 which one has to use a lot in wizards in the indices 0 to 19 of an array. But of course, this would only "work" if addressing array elements directly is possible (which is something that's actually ugly).

### LINKED ISSUE #338: No static imports
Static imports of fields shall not be allowed. One could further restrict static imports to some methods.

### LINKED ISSUE #345: [Checkstyle] Assert Javadoc formatting
* Order of tags
* Newlines
* Formatting of HTML

### LINKED ISSUE #346: [Checkstyle] Configuration for Prototypes
The same as main Beagle but

 * Allow sysout
 * Allow main methods
 * Allow TODO

### LINKED ISSUE #347: [Checkstyle] Adapt Checkstyle regarding Array Creation
Checkstyle and the formatter should both require that there are no blanks before the braces in
> final String[] buttonLabels = {"Abort", "Pause"};

### PR DESCRIPTION
**Pull request by acBerger**
fixes #317 
fixes #338 
fixes #318 
fixes #320 
fixes #324 
fixes #332 
fixes #326 
fixes #327
fixes #347  
fixes #334
fixes #328 
fixes #345
fixes #346

[Review on Reviewable](https://reviewable.io/reviews/beagle-pse/beagle/348)

### CODE REVIEW

**Comment by m1cm1c** [comment]
Introduces this new conflict between Checkstyle and formatter:

**Comment by m1cm1c** [comment]
Just in case you want to know what the new formatter did differently here:

**Comment by m1cm1c** [comment]
> fixes #328

Nope. I still get the same warning.

**Comment by m1cm1c** [comment]
> fixes #347

Neither Checkstyle nor the formatter changed regarding this issue.

**Comment by acBerger** [comment]
> Neither Checkstyle nor the formatter changed regarding this issue.

Yes, it does. 
see 
 - line 160 and 191 in  .settings/org.eclipse.jdt.core.prefs 
- line 49 and 192 in  config/Formatter.xml

**Comment by m1cm1c** [comment]
>> Neither Checkstyle nor the formatter changed regarding this issue.

>Yes, it does.
> see

>    line 160 and 191 in .settings/org.eclipse.jdt.core.prefs
>    line 49 and 192 in config/Formatter.xml

You're right. You changed the location of the file and my Eclipse still used the old formatter.

**Comment by m1cm1c** [comment]
But I still get the warning of #328.

**Comment by acBerger** [comment]
Yes, the location was changed in order to use checkstyle in gradle.

> But I still get the warning of #328 

It worked for my example but I'll try to fix it. Probably they are a bit different.

**Comment by m1cm1c** [comment]
Checkstyle now even wants to break the line `if (TimeoutPage.this.textboxTimeoutSeconds.getText().isEmpty()) {` (16 characters of indentation + 65 characters code) before `.isEmpty())`. This is way too short of a line.

**Comment by m1cm1c** [comment]
>>But I still get the warning of #328

> It worked for my example but I'll try to fix it. Probably they are a bit different.

You can try it with my branch `eclipse-gui-prototype-and-the-tiny-wizard-of-dialogia`.

**Comment by jGleitz** [comment]
@m1cm1c: did you remember to re-import the `Formatter.xml`? Does it work for you now?

**Comment by m1cm1c** [comment]
Oh, I didn't notice @acBerger pushed commits.

Yes, it works. However, the main method still doesn't match the Checkstyle rules (I tried the normal version as well as the prototype one).

File: config/checkstyle/checkstyle Prototyp.xml [hunk 1]
```diff
@@ -0,0 +1,579 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (m1cm1c) [inline config/checkstyle/checkstyle Prototyp.xml]: The file name should be in English and follow our file name convention.

File: config/checkstyle/checkstyle.xml [hunk 2]
```diff
@@ -78,7 +78,7 @@
     </module>
     <module name="AbstractClassName">
       <property name="severity" value="warning"/>
-      <property name="format" value="^[A-Z][A-Za-z]{5,}$"/>
+      <property name="format" value="^[A-Z][A-Za-z]{3,}$"/>
```
Comment (m1cm1c) [inline config/checkstyle/checkstyle.xml]: The file name should follow our file name convention.

**Comment by jGleitz** [comment]
@acBerger Please update the branch

**Comment by RomanLangrehr** [comment]
The import order should have the file name extension `.importorder`

**Comment by RomanLangrehr** [comment]
image

Message: `'(' sollte in einer neuen Zeile stehen.`

Me and the formatter disagree on placing `(` in a new line.

**Comment by acBerger** [comment]
This should fix both

**Comment by m1cm1c** [comment]
I still get the error
> Name 'Main' must match pattern '^[A-Z][a-zA-Z0-9]{7,}$'.

**Comment by acBerger** [comment]
Are you using the prototype checkstyle?
Am 31.12.2015 11:20 schrieb "m1cm1c" <notifications@github.com>:

**Comment by m1cm1c** [comment]
> Are you using the prototype checkstyle?

Yes. And I made sure to use the correct file (since you corrected the file name).

**Comment by acBerger** [comment]
@m1cm1c Sorry, seems like I forgot this one. I added 'Main' to the allowed Type Names for Prototypes.

**Comment by m1cm1c** [comment]
Thank you. lgtm

**Comment by jGleitz** [comment]
Reviewed 2 of 6 files at r2, 2 of 6 files at r3, 1 of 3 files at r4, 3 of 4 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[.settings/org.eclipse.jdt.core.prefs, line 11 \[r6\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sYjEINMjcoBbs-pEZ:-K6sYjEJy2e_lD1SfVTQ:41578430) ([raw file](https://github.com/beagle-pse/beagle/blob/48d1ab7b6cd81aebf297c46518de7d192b068315/.settings/org.eclipse.jdt.core.prefs#L11)):*
Do you know what these lines do?

---

*[.settings/org.eclipse.jdt.ui.prefs, line 63 \[r6\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sZ544_T-RxQ4vMoJW:-K6sZ544_T-RxQ4vMoJX:-1403227008) ([raw file](https://github.com/beagle-pse/beagle/blob/48d1ab7b6cd81aebf297c46518de7d192b068315/.settings/org.eclipse.jdt.ui.prefs#L63)):*
Does this mean how exception variables will be named in autogenerated `catch` blocks? If yes, we could change that to `exception` the stubs would then obey the checkstyle rules.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/348)*

**Comment by acBerger** [comment]
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[.settings/org.eclipse.jdt.core.prefs, line 11 \[r6\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sYjEINMjcoBbs-pEZ:-K6sbvlGBHZxzR00taDC:1904615773) ([raw file](https://github.com/beagle-pse/beagle/blob/48d1ab7b6cd81aebf297c46518de7d192b068315/.settings/org.eclipse.jdt.core.prefs#L11)):*
No, they were changed automatically when I did some changes on the checkstyle files in eclipse.

---

*[.settings/org.eclipse.jdt.ui.prefs, line 63 \[r6\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sZ544_T-RxQ4vMoJW:-K6sbOSqmD6sYMxHzkkP:528479477) ([raw file](https://github.com/beagle-pse/beagle/blob/48d1ab7b6cd81aebf297c46518de7d192b068315/.settings/org.eclipse.jdt.ui.prefs#L63)):*
I did this on purpose as it is very easy forgotten to change the name of the exception to a more meaningful one, if there is no warning. On the other hand I cannot forbid calling them exception as this could be the right term sometimes.

---

*[config/checkstyle/Checkstyle Prototype.xml, line 1 \[r2\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sRrB6EHlo0tuCk_0p-r2-1:-K6scN8Kj0kzsCTBQ2K7:-499146009) ([raw file](https://github.com/beagle-pse/beagle/blob/906850b916b494e91599c49f01e172cc11655ba5/config/checkstyle/Checkstyle Prototype.xml#L1)):*
Done.

---

*[config/checkstyle/Checkstyle.xml, line 81 \[r2\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sRrB6EHlo0tuCk_0o-r2-81:-K6scPnSPwrXMV6mNvWm:-499146009) ([raw file](https://github.com/beagle-pse/beagle/blob/906850b916b494e91599c49f01e172cc11655ba5/config/checkstyle/Checkstyle.xml#L81)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/348)*

**Comment by jGleitz** [comment]
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[.settings/org.eclipse.jdt.ui.prefs, line 63 \[r6\]](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-K6sZ544_T-RxQ4vMoJW:-K6scvZdJhHT2LZT4wpd:1776681207) ([raw file](https://github.com/beagle-pse/beagle/blob/48d1ab7b6cd81aebf297c46518de7d192b068315/.settings/org.eclipse.jdt.ui.prefs#L63)):*
I see.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/348)*

**Comment by jGleitz** [comment]
I think @RomanLangrehr and @m1cm1c should approve this as they have “real” code to test on.

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-:-K6sdClkXZ0MM_hRnZdX:-1204240485)*

**Comment by m1cm1c** [comment]
:lgtm: 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/348#-:-K6spbj9rxZ5NkOiSoPm:980077423)*

**Comment by m1cm1c** [comment]
Somehow the first "lgtm" (edited by @jGleitz) didn't count. Here's another one.

**Comment by jGleitz** [comment]
Reviewed 1 of 3 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/beagle-pse/beagle/348)*

**Comment by RomanLangrehr** [comment]
:lgtm:

**Pull request merged**